tap_klaviyo/utils.py

```diff
@@ -3,13 +3,70 @@
 import singer
 from singer import metrics
 import requests
+import backoff
+import json
+import simplejson
 
 DATETIME_FMT = "%Y-%m-%dT%H:%M:%SZ"
 
-
 session = requests.Session()
 logger = singer.get_logger()
 
+class KlaviyoError(Exception):
+    pass
+
+class KlaviyoNotFoundError(KlaviyoError):
+    pass
+
+class KlaviyoBadRequestError(KlaviyoError):
+    pass
+
+class KlaviyoUnauthorizedError(KlaviyoError):
+    pass
+
+class KlaviyoForbiddenError(KlaviyoError):
+    pass
+
+class KlaviyoInternalServiceError(KlaviyoError):
+    pass
+
+ERROR_CODE_EXCEPTION_MAPPING = {
+    400: {
+        "raise_exception": KlaviyoBadRequestError,
+        "message": "Request is missing or has a bad parameter."
+    },
+    401: {
+        "raise_exception": KlaviyoUnauthorizedError,
+        "message": "Invalid authorization credentials."
+    },
+    403: {
+        "raise_exception": KlaviyoForbiddenError,
+        "message": "Request is missing or has an invalid API key."
```

Comment:
Can we adjust the message here to say "Invalid authorization credentials or permissions."

Reply:
I have added the desired string, however I have taken the string from Klaviyo Documentation https://www.klaviyo.com/docs/api/lists.
So, let me know if want to revert the change.